Automattic/newspack-newsletters
# Add service provider picking

### All Submissions:






### Changes proposed in this Pull Request:



#274 introduced handliing Constant Contact as an ESP, this PR adds UI to choose/change an ESP.

### How to test the changes in this Pull Request:

1. If you already had Mailchimp provider set up, all should work as before. If starting afresh, first set up Mailchimp on `master` branch and verify that the Mailchimp connection works as before.
2. Remove Mailchimp credentials in plugin settings (or start on a fresh site)
3. Observe the Settings view has a ESP provider selector and three credentials inputs – one for Mailchimp and two for Constant Contact
4. Create a new newsletter – observe a settings modal
5. Fill in the settings, setting up Mailchimp as the provider
6. Observe Mailchimp connection working
7. Switch the service provider to Constant Contact
7. Observe Constant Contact connection working (campaign created on newsletter creation)
7. Start from scratch (remove all options' values of start on a fresh site) on this branch, set up Constant Contact
7. Observe Constant Contact connection working
7. Switch the service provider to Mailchimp
6. Observe Mailchimp connection working

### Other information:
